juspay/hyperswitch
# feat(business_profile): feature add a config to use `billing` as `payment_method_billing`

## Type of Change



## Description

This PR adds support for a config which controls whether to use the `billing` as `payment_method_billing`. Based on this config the address is unified and accessed at the connector module. This also affects the payment method list to either populate or not populate the billing fields.

### Additional Changes

A new field called `use_billing_as_payment_method_billing` is added to business profile create and update.

A new field called `use_billing_as_payment_method_billing` is added to business profile table.




## Motivation and Context


## Impact
- Passing billing address to connectors.
- Automatic filling of billing details in payment method list.

## How did you test it?

- Create a payment with default config, by passing `billing` details in payment.
```bash
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'api-key: dev_3Ehos25h2Ro9Y6jh7wz3RTQHm4Hfk7xenFpl56S1BPZG3YsTzK8j1YHD9XhBdI46' \
--data-raw '{
    "amount": 6540,
    "currency": "USD",
    "confirm": true,
    "customer": {
        "id": "cus_abc",
        "email": "example@juspay.in"
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "Narayan",
            "last_name": "Bhat"
        }
    }
}'
```

- The SDK will not collect any billing data ( for cybersource connector )
<img width="872" alt="image" src="https://github.com/juspay/hyperswitch/assets/48803246/da4b2dc4-81bc-4ced-8cf8-4bc99d08d91d">

- Disable the config for the business profile
```bash
curl --location 'http://localhost:8080/account/merchant_1714987557/business_profile/pro_nEF5mTQDHBMNbJejr85j' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
  "use_billing_as_payment_method_billing": false
}'
```

- Now the SDK should collect all billing details from customer
<img width="874" alt="image" src="https://github.com/juspay/hyperswitch/assets/48803246/fd32b5ef-a68f-4d69-8f90-e92f3dd2af2d">


## Checklist
